### DISCUSSION TITLE
Recent Podcast list to Pushover(android etc) using python

### DISCUSSION THREAD

**Discussion by theorbitofmercury** [dvehrs/podget#60] [Show and tell]
In case this is of use to anyone. I wanted a list of recent e.g. last 48h podcasts as a list in Pushover Android app(also available for IoS  and Desktops). Then I can easily see what new podcasts have been downloaded.
https://pushover.net/
Android : https://play.google.com/store/apps/details?id=net.superblock.pushover

This is my python3 code running on a Raspberry OS Pi 4 (buster) where the podcasts are downloaded:
```
#!/usr/bin/env python3
#https://stackoverflow.com/questions/16953842/using-os-walk-to-recursively-traverse-directories-in-python
#given a folder name, walk through its entire hierarchy

import os, time, http.client, urllib
now = time.time()
nowLess24h = time.time() - (48*60*60) #48 hours
podList = []

def recursive_walk(folder):
    global podList
    for folderName, subfolders, filenames in os.walk(folder):
        if subfolders:
            for subfolder in subfolders:
                recursive_walk(subfolder)
        created = os.path.getctime(folderName)
        if created <= now and created >= nowLess24h:
            #print(folderName, created)
            lenFolderName = len(folderName) 
            folderName = (folderName[13:lenFolderName])
            if folderName != ".LOG" and folderName != "m3u-tmp":
                #print(folderName)
                podList.append(folderName)

recursive_walk('/home/pi/POD/') #where your podcast's are located)

podsStr = "<font color='#ff8000'><b>Last 48h pod Casts</b></font>" + chr(int(10)) + ''.join(str(e) + chr(int(10)) for e in podList) #the font etc stuff is to give a coloured heading to the Pushover list of podcasts
#print(podsStr)
PUSH_MSG =  podsStr

#setup pushover API
app_key = "xxx"
user_key = "yyy"

#This function sends the push message using Pushover.
def sendPush(PUSH_MSG):
   conn = http.client.HTTPSConnection("api.pushover.net:443")
   conn.request("POST", "/1/messages.json",
      urllib.parse.urlencode({
         "token": app_key,
         "user": user_key,
         "message": PUSH_MSG,
         "html": 1,
         "sound": "tugboat"
      }), { "Content-type": "application/x-www-form-urlencoded" })

   conn.getresponse()
   return
   
# Run function to send message to Pushover
sendPush(PUSH_MSG)
```
I've put this in cron for twice per day after my twice per day podget's
Cheers Mike

**Comment by dvehrs** [comment]
Very interesting little python script.

If you want, we can add this to the SCRIPTS directory in the Podget repository.  Now I could add it myself but then you wouldn't get credit as a contributor.  So I'd suggest that you fork a copy of Podget then create a v0.8.8 directory in the SCRIPTS folder to hold the file and a README for telling people how to use it.   Once you have it the way you like it, create a pull request to Podget and then we can integrate it.   

Dave

**Reply by theorbitofmercury** [reply]
I've done as you suggested. Hope the github procedure is right as it new territory for me.
Cheers Mike

**Reply by dvehrs** [reply]
OK, I've accepted your pull request into the DEV branch for inclusion with the next release.  As we discussed I also renamed the file from README to push-new-mp3.REAME.   Only one thing left to do and that is add the COPYRIGHT/Author lines to the files.   Unfortunately i don't see enough in your profile to have a good guess as to what you would like them to be.  I imagine something like "Copyright (c) 2021 Mike (lastname)", and same for Author credits in the README.  What would you like it to be?  

Thanks,
Dave

**Reply by theorbitofmercury** [reply]
Hi Dave
 "Copyright (c) 2021 Mike Webb" is fine for both the code and the README to push-new-mp3.
Cheers
Mike
